src/Core/Query/AbstractDocument.php

```diff
@@ -24,10 +24,8 @@ abstract class AbstractDocument implements DocumentInterface, \IteratorAggregate
     /**
      * @param mixed $name
      * @param mixed $value
-     *
-     * @return \Solarium\Core\Query\DocumentInterface
      */
-    abstract public function __set($name, $value): DocumentInterface;
+    abstract public function __set($name, $value): void;
```

Comment:
Can this be a BC break. Not sure if anyone can be relying on the return type here. Maybe yes.

But it definitely was invalid even on php7 and should have never worked like this.

See https://www.php.net/manual/en/language.oop5.overloading.php#object.set



WDYT?

Reply:
According to the documentation this is rather a bug fix then a BC break.